src/ClientTransactionManager.cc

```diff
@@ -0,0 +1,76 @@
+/* Copyright (c) 2015 Stanford University
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include "RamCloud.h"
+#include "ClientTransactionManager.h"
+#include "ClientTransactionTask.h"
+
+namespace RAMCloud {
+
+/**
+ * Constructor for the client transaction manager.
+ *
+ * \param ramcloud
+ *      Overall information about the client.
+ */
+ClientTransactionManager::ClientTransactionManager(RamCloud* ramcloud)
```

Comment:
I don't understand why this class is needed. Why don't you just create a separate timer handler (or, better yet, poller) for each  active TransactionTask?